From 149319819aa5d4e39770c35a0c00f6c2963e2bf3 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Sat, 20 Feb 2021 21:02:05 +0100
Subject: [PATCH 27/46] jack: apply PIPEWIRE_PROPS after reading config

First apply config properties, then PIPEWIRE_PROPS.
We can set the node.latency in jack.conf
---
 pipewire-jack/src/pipewire-jack.c | 20 ++++++++++----------
 src/daemon/jack.conf.in           |  1 +
 2 files changed, 11 insertions(+), 10 deletions(-)

diff --git a/pipewire-jack/src/pipewire-jack.c b/pipewire-jack/src/pipewire-jack.c
index 3bf0af1c8..90895e476 100644
--- a/pipewire-jack/src/pipewire-jack.c
+++ b/pipewire-jack/src/pipewire-jack.c
@@ -2421,19 +2421,16 @@ jack_client_t * jack_client_open (const char *client_name,
 	varargs_parse(client, options, ap);
 	va_end(ap);
 
-        if ((str = getenv("PIPEWIRE_PROPS")) != NULL)
-		client->props = pw_properties_new_string(str);
-	if (client->props == NULL)
-		client->props = pw_properties_new(NULL, NULL);
+	client->props = pw_properties_new(
+			"loop.cancel", "true",
+			PW_KEY_REMOTE_NAME, client->server_name,
+			PW_KEY_CLIENT_NAME, client_name,
+			PW_KEY_CLIENT_API, "jack",
+			PW_KEY_CONFIG_NAME, "jack.conf",
+			NULL);
 	if (client->props == NULL)
 		goto no_props;
 
-	pw_properties_set(client->props, "loop.cancel", "true");
-	pw_properties_set(client->props, PW_KEY_REMOTE_NAME, client->server_name);
-	pw_properties_set(client->props, PW_KEY_CLIENT_NAME, client_name);
-	pw_properties_set(client->props, PW_KEY_CLIENT_API, "jack");
-	pw_properties_set(client->props, PW_KEY_CONFIG_NAME, "jack.conf");
-
 	client->node_id = SPA_ID_INVALID;
 	strncpy(client->name, client_name, JACK_CLIENT_NAME_SIZE);
 	client->context.loop = pw_thread_loop_new(client_name, NULL);
@@ -2452,6 +2449,9 @@ jack_client_t * jack_client_open (const char *client_name,
 					"jack.properties")) != NULL)
 		pw_properties_update_string(client->props, str, strlen(str));
 
+        if ((str = getenv("PIPEWIRE_PROPS")) != NULL)
+		pw_properties_update_string(client->props, str, strlen(str));
+
 	if ((str = pw_properties_get(client->props, "jack.merge-monitor")) != NULL)
 		client->merge_monitor = pw_properties_parse_bool(str);
 
diff --git a/src/daemon/jack.conf.in b/src/daemon/jack.conf.in
index dec2d5c71..2ef4877ad 100644
--- a/src/daemon/jack.conf.in
+++ b/src/daemon/jack.conf.in
@@ -54,5 +54,6 @@ context.modules = {
 }
 
 jack.properties = {
+     #node.latency = 1024/48000
      #jack.merge-monitor  = false
 }
-- 
2.26.2

